src/Struct/FindologicProduct.php

```diff
@@ -325,6 +325,50 @@ protected function setUrl(): void
         $this->url = $productUrl;
     }
 
+    protected function getTranslatedSeoPath(): ?string
+    {
+        $salesChannel = $this->salesChannelContext->getSalesChannel();
+        $seoUrlCollection = $this->product->getSeoUrls()->filterBySalesChannelId($salesChannel->getId());
+
+        /** @var SeoUrlEntity|null $seoUrlEntity */
+        $seoUrlEntity = $this->getTranslatedEntity($seoUrlCollection);
+
+        return $seoUrlEntity ? $seoUrlEntity->getSeoPathInfo() : $seoUrlEntity;
+    }
+
+    protected function getTranslatedDomainBaseUrl(): ?string
+    {
+        $salesChannel = $this->salesChannelContext->getSalesChannel();
+        $domainCollection = $salesChannel->getDomains();
+
+        /** @var SalesChannelDomainEntity|null $domainEntity */
+        $domainEntity = $this->getTranslatedEntity($domainCollection);
+
+        return $domainEntity ? $domainEntity->getUrl() : $domainEntity;
+    }
+
+    /**
+     * Finds the first entity of a collection for the export language and returns it. If none is found,
+     * null is returned.
+     */
+    protected function getTranslatedEntity(?EntityCollection $collection): ?Entity
+    {
+        if (!$collection) {
+            return null;
+        }
+
+        $translatedEntities = $collection->filterByProperty(
+            'languageId',
+            $this->salesChannelContext->getSalesChannel()->getLanguageId()
+        );
+
+        if ($translatedEntities->count() === 0) {
+            return null;
+        }
+
+        return $translatedEntities->first();
```

Comment:
Possible to have more than one translatedEntitites?

Reply:
Theoretically yes, but practically no, as the SEO URLs are generated by Shopware via the indexer.